### PR TITLE
Fix Jackalope\Query\Row create bug

### PR DESCRIPTION
**Pull request by middlebrain**
The following code

```php
...
$ws = $session->getWorkspace();
$qm = $ws->getQueryManager();

$result = $qm->createQuery(
  "SELECT [jcr:primaryType] FROM [rep:root]",
  "JCR-SQL2"
)
->execute();

foreach ($result as $row) {
  echo $row->getValue("jcr:primaryType") . "\n";
}
```
throws
```
[PHPCR\ItemNotFoundException]
Column '.jcr:primaryType' not found
```

### CODE REVIEW

File: src/Jackalope/Query/Row.php [hunk 1]
```diff
@@ -107,7 +107,7 @@ public function __construct(FactoryInterface $factory, ObjectManager $objectMana
                 $this->path[$selectorName] = $column['dcr:value'];
             } else {
                 if ('jcr:primaryType' === substr($column['dcr:name'], -15)) {
-                    $this->defaultSelectorName = $selectorName;
+                    $this->defaultSelectorName = $column['dcr:value'];
```
Comment (dbu) [inline src/Jackalope/Query/Row.php]: is this change related to the fix?

Comment (middlebrain) [reply src/Jackalope/Query/Row.php]: No. I actually wanted to make two pull requests of my two commits, but I lost the fight with GitHub. This is the commit of my "other pull request" (see https://github.com/jackalope/jackalope/pull/351/commits/018f1968c77d0a7e099b27d301122f87534e645b). I have no clue how to fix my mistake.

Comment (dbu) [reply src/Jackalope/Query/Row.php]: you can checkout your fork, create a new branch from master and cherry-pick that second commit `git cherry-pick 018f196`. then go to this branch and `git rebase -i origin/master` and remove the second commit. that will leave only the first commit in the branch. then `git push --force-with-lease` to push the change upstream.

**Comment by dbu** [comment]
the changelog entry looks good to me.

**Comment by dbu** [comment]
i split the two commits and merged them on the cli, and updated the changelog. will release as 1.3.6 in a few minutes. thanks a lot for the fixes!

**Comment by middlebrain** [comment]
Thank you very much for the resolution of the chaos I caused. Unfortunately, I had not yet found time to follow your advice to resolve it myself.

**Comment by dbu** [comment]
no worries, finding the bugs and solutions for them was the important part of the work. thanks again!

File: src/Jackalope/Query/Row.php (hunk 1 above)
Comment (alexander-schranz) [reply src/Jackalope/Query/Row.php]: this line does currently break our build in sulu/sulu on jackrabbit `PHPCR\RepositoryException: Attempting to get the path for a non existent selector: nt:unstructured`. doctrine dbal seems to work fine. any ideas?

/cc @dbu @middlebrain

Comment (alexander-schranz) [reply src/Jackalope/Query/Row.php]: Running the test in jackalope-jackrabbit errors with:

```
1) PHPCR\Tests\Query\QuerySql2OperationsTest::testQueryFieldDate
PHPCR\RepositoryException: Attempting to get the path for a non existent selector: nt:unstructured

/private/tmp/jackalope-jackrabbit/vendor/jackalope/jackalope/src/Jackalope/Query/Row.php:208
/private/tmp/jackalope-jackrabbit/vendor/phpcr/phpcr-api-tests/tests/Query/QuerySql2OperationsTest.php:70

ERRORS!
Tests: 1866, Assertions: 8624, Errors: 1, Skipped: 62, Incomplete: 12.
```
